docs: add a guide to blog.md (#118)
asyncapi/.github: .github/ISSUE_TEMPLATE/blog.md

---

<!--

> We're super happy to see you want to publish on our blog. The AsyncAPI blog is open for everyone, any contribution is allowed, including reposting.
> By posting an article on the AsyncAPI blog, you get many advantages. We'll ask on the community Slack to help promote it whenever it's' released. We'll share it on the official AsyncAPI Twitter, LinkedIn, Hacker News, and Reddit.
> You can share your use cases, products, or tools in a form of a blog post to reach out to the entire AsyncAPI community.
> Please fill in the data below and we'll guide you through the process.

Reviewer: ```suggestion
We're super happy to see you want to publish on our blog. The AsyncAPI blog is open for everyone; any contribution is allowed, including reposting.
By posting an article on the AsyncAPI blog, you get many advantages. We'll ask on the community Slack to help promote it whenever released. We'll share it on the official AsyncAPI Twitter, LinkedIn, Hacker News, and Reddit.
You can even share your use cases, products, or tools in a blog post to reach out to the entire AsyncAPI community.
Please fill in the data below, and we'll guide you through the process. 
```

If you already published the article on another blog that's
fine too. Please provide the link to the article.
-->

## Guide
Here are a few steps you can follow to write an AsyncAPI blog post:

> 1. After getting feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website) or open it in Gitpod. You can do that by prefixing the issue URL with gitpod.io/#.

Reviewer 2: ```suggestion
1. After getting feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website) or open it in Gitpod. You can do that by prefixing the issue URL with `gitpod.io/#`.
```

If you already published the article on another blog that's
fine too. Please provide the link to the article.
-->

## Guide
Here are a few steps you can follow to write an AsyncAPI blog post:

1. After getting feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website).
1. Run the command `npm run write:blog`.

> 1. Run the website locally or with Gitpod by using instructions from the `README` to test your changes

Reviewer: ```suggestion
1. Run the website locally or with Gitpod by using instructions from the `README` to test your changes.
```

> 1. Run the website locally or with Gitpod by using instructions from the `README` to test your changes.

Reviewer 3: When I fork, and generate file locally, I can't really test it with GitPod, right? or?

Author: @derberg ahh right, if you fork you can't test with gitpod 😅.
Perhaps we should just remove the Gitpod part wdyt? Or do `After getting feedback on the issue, fork the website[repository](https://github.com/asyncapi/website). or use Gitpod to get a development environment`

Reviewer: I'm unclear on this too! We should ask @mikenikles or @pawlean who work at Gitpod and can explain how this works for sure 😄

Reviewer 4: I'd write something along the lines of:

1. After getting feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website) or open it in Gitpod. You can do that by prefixing the issue URL with `gitpod.io/#`.
1. Run the command `npm run write:blog`.
1. Run the website by using instructions from the `README` to test your changes
1. Open a PR with your blog post and test your changes with a preview of the site on Netlify. If you use Gitpod, it will create a fork of the `website` repo for you before you create a PR.

Reviewer: Thank you bunches @mikenikles!

Author: thank you all for the help 👍 , it's ready for review again 🙂

> Here are a few steps you can follow to write a blog post for AsyncAPI:
> 
> 1. After getting some feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website)
> 1. Run the command `npm run write:blog`
> 1. Run the website locally or with Gitpod by using instructions from the `README` to test your changes
> 1. Open a PR with your blog post and test your changes with a preview of the site on Netlify.
> 1. Maintainers will review the article (e.g. language, images) and potentially ask for improvements
> 1. Once it gets merged it will be available live in production :rocket:
> 
> We encourage you to write a blog post and sharing it with the community. We can't wait to read it :smile:!

Reviewer: ```suggestion
Here are a few steps you can follow to write an AsyncAPI blog post:

1. After getting feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website).
1. Run the command `npm run write:blog`.
1. Run the website locally or with Gitpod by using instructions from the `README` to test your changes
1. Open a PR with your blog post and test your changes with a preview of the site on Netlify.
1. Maintainers reviewing the article (e.g., language, images) may ask for improvements.
1. Once it's merged, it will be available live in production. :rocket:

We encourage you to write a blog post and share it with the community. We can't wait to read it :smile:!
```

If you already published the article on another blog that's
fine too. Please provide the link to the article.
-->

## Guide
Here are a few steps you can follow to write a blog post for AsyncAPI:

1. After getting some feedback on the issue, fork the `website` [repository](https://github.com/asyncapi/website)

> 2. Add a new markdown file [here](https://github.com/asyncapi/website/tree/master/pages/blog) with proper metadata (front matter) and the article content, as other posts have. The content of the article is up to you, it just needs to mention "AsyncAPI"

Reviewer 3: shouldn't instruction just say `npm post:add`? and the rest of the content that is actually related to content of the blog post should maybe be added inside the generated post file?

could be better approach as we can then put more stuff there, showcase how youtube links should be added, showcase how links should be added, showcase how tweets can be embedded, basically the generated blog post would contain bunch of examples user can learn from and remove if not needed

thoughts? then we maintain majority of instructions inside website, no need to update here when new things happen

Author: yup totally agreed, thanks I'll make those changes 👍

> 3. Add a cover image. In case you need some inspiration we recommend https://unsplash.com/. All images should be placed [here](https://github.com/asyncapi/website/tree/master/public/img/posts) into a folder that has the same name as the markdown file created a step before. Also, make sure to have the following:

Reviewer 3: please add user should make sure to give credit to source of image

Author: I'll add that notice on the `website` repo's PR, for the reason you mentioned on the other comment 🙂
